src/compute-client/src/controller.rs

```diff
@@ -489,6 +533,106 @@ where
         Ok(())
     }
 
+    /// Create and maintain the described dataflows, and initialize state for their output.
+    ///
+    /// This method creates dataflows whose inputs are still readable at the dataflow `as_of`
+    /// frontier, and initializes the outputs as readable from that frontier onward.
+    /// It installs read dependencies from the outputs to the inputs, so that the input read
+    /// capabilities will be held back to the output read capabilities, ensuring that we are
+    /// always able to return to a state that can serve the output read capabilities.
+    pub async fn create_dataflows(
+        &mut self,
+        instance_id: ComputeInstanceId,
+        dataflows: Vec<DataflowDescription<crate::plan::Plan<T>, (), T>>,
+    ) -> Result<(), ComputeError> {
+        self.instance(instance_id)?
+            .create_dataflows(dataflows)
+            .await
+    }
+
+    /// Drop the read capability for the given collections and allow their resources to be
+    /// reclaimed.
+    pub async fn drop_collections(
+        &mut self,
+        instance_id: ComputeInstanceId,
+        collection_ids: Vec<GlobalId>,
+    ) -> Result<(), ComputeError> {
+        self.instance(instance_id)?
+            .drop_collections(collection_ids)
+            .await
+    }
+
+    /// Initiate a peek request for the contents of the given collection at `timestamp`.
+    pub async fn peek(
+        &mut self,
+        instance_id: ComputeInstanceId,
+        collection_id: GlobalId,
+        literal_constraints: Option<Vec<Row>>,
+        uuid: Uuid,
+        timestamp: T,
+        finishing: RowSetFinishing,
+        map_filter_project: mz_expr::SafeMfpPlan,
+        target_replica: Option<ReplicaId>,
+    ) -> Result<(), ComputeError> {
+        self.instance(instance_id)?
+            .peek(
+                collection_id,
+                literal_constraints,
+                uuid,
+                timestamp,
+                finishing,
+                map_filter_project,
+                target_replica,
+            )
+            .await
+    }
+
+    /// Cancel existing peek requests.
+    ///
+    /// Canceling a peek is best effort. The caller may see any of the following
+    /// after canceling a peek request:
+    ///
+    ///   * A `PeekResponse::Rows` indicating that the cancellation request did
+    ///    not take effect in time and the query succeeded.
+    ///
+    ///   * A `PeekResponse::Canceled` affirming that the peek was canceled.
+    ///
+    ///   * No `PeekResponse` at all.
```

Comment:
(same here) Can you add in the docstring in which cases this method can fail?

I think it's in particular important to understand if the only failure mode is invalid id or if there are also other cases, that gives the caller some confidence whether to call `.unwrap()` or not